From: ddugger@redhat.com <ddugger@redhat.com>
Date: Mon, 6 Apr 2009 15:51:42 -0600
Subject: [misc] xen dom0: add hypercall for add/remove PCI device
Message-id: 200904062151.n36LpgBC013508@sobek.n0ano.com
O-Subject: [RHEL5.4 PATCH 2/3] BZ484227: VTD: dom0: add hypercall definition for adding and removing PCI devices
Bugzilla: 484227
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>

sync with hypercall definitions in xen/include/public/physdev.h,
then the hypercall can be called from dom0.

Upstream Status: Accepted (CS 592)

Signed-off-by: Weidong Han <weidong.han@intel.com>
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Signed-off-by: Don Dugger <donald.d.dugger@intel.com>

diff --git a/include/xen/interface/physdev.h b/include/xen/interface/physdev.h
index 03e3413..6798d95 100644
--- a/include/xen/interface/physdev.h
+++ b/include/xen/interface/physdev.h
@@ -84,6 +84,17 @@ struct physdev_apic {
 typedef struct physdev_apic physdev_apic_t;
 DEFINE_XEN_GUEST_HANDLE(physdev_apic_t);
 
+#define PHYSDEVOP_manage_pci_add         15
+#define PHYSDEVOP_manage_pci_remove      16
+struct physdev_manage_pci {
+    /* IN */
+    uint8_t bus;
+    uint8_t devfn;
+};
+
+typedef struct physdev_manage_pci physdev_manage_pci_t;
+DEFINE_XEN_GUEST_HANDLE(physdev_manage_pci_t);
+
 /*
  * Allocate or free a physical upcall vector for the specified IRQ line.
  * @arg == pointer to physdev_irq structure.
